src/components/CustomInput/index.js

```diff
@@ -0,0 +1,24 @@
+import PropTypes from "prop-types";
+import TextField from "@mui/material/TextField";
+
+const CustomInput = ({ value, type, name, placeholder, onChange }) => {
+  return (
+    <TextField
+      defaultValue={value}
+      name={name}
+      type={type}
+      onChange={onChange}
+      placeholder={placeholder}
```

Comment:
I don't see the value prop to input. 
add `value={value}` to TextField component. That's the issue why your form isn't cleaned up